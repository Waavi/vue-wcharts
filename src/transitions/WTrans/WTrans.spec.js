import { mount, shallowMount } from '@vue/test-utils'
import WTrans from './WTrans.vue'

jest.useFakeTimers()

describe('Widgets/WTrans', () => {
    const propsData = {
        id: 'test',
        initialProps: {
            opacity: 0,
        },
        transition: 'all 250s 250ss ease',
    }

    const defaultConfig = {
        context: {
            props: propsData,
        },
    }

    xit(`Should be render correctly`, () => {
        const wrapper = shallowMount(WTrans, defaultConfig)
        expect(wrapper.html()).toMatchSnapshot()
    })

    xit(`Should be render correctly with children`, () => {
        const wrapper = mount({
            components: { WTrans },
            props: {
                initialProps: {
                    opacity: 0,
                },
                transition: 'all 250s 250ss ease',
            },
            template: `
              <div>
                <WTrans :initialProps="initialProps" :transition="transition">
                    <span>Hello</span>
                </WTrans>
              </div>
            `,
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    xit(`Should be render correctly with children`, (done) => {
        const h = (tag, { props, on }) => {
            const el = {
                getAttribute: (key) => {},
                setAttribute: (key, value) => {},
                style: {},
            }

            expect(tag).toEqual('transition')
            expect(props.appear).toBeTruthy()

            on.beforeAppear(el)
            on.appear(el)

            expect(setTimeout).toHaveBeenCalledTimes(1)
            done()
        }
        expect(WTrans.render(h, { props: propsData })).toMatchSnapshot()
    })
})
