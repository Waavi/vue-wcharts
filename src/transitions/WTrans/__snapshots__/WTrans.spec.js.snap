// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/WTrans Should be render correctly 1`] = `undefined`;

exports[`Widgets/WTrans Should be render correctly with children 1`] = `<div><span>Hello</span></div>`;

exports[`Widgets/WTrans Should be render correctly with children 2`] = `undefined`;
